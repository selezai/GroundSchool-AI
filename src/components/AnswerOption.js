import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

const AnswerOption = ({
  option,
  isSelected,
  isCorrect,
  showResult,
  onSelect,
  testID = 'answer-option'
}) => {
  const getStyles = () => {
    if (!showResult) {
      return isSelected ? styles.selected : styles.default;
    }
    if (isCorrect) {
      return styles.correct;
    }
    return isSelected ? styles.incorrect : styles.default;
  };

  const getTextStyles = () => {
    if (!showResult) {
      return isSelected ? styles.selectedText : styles.defaultText;
    }
    if (isCorrect) {
      return styles.correctText;
    }
    return isSelected ? styles.incorrectText : styles.defaultText;
  };

  return (
    <TouchableOpacity
      style={[styles.container, getStyles()]}
      onPress={onSelect}
      disabled={showResult}
      testID={testID}
    >
      <Text style={[styles.text, getTextStyles()]}>
        {option}
      </Text>
    </TouchableOpacity>
  );
};

AnswerOption.propTypes = {
  option: PropTypes.string.isRequired,
  isSelected: PropTypes.bool.isRequired,
  isCorrect: PropTypes.bool.isRequired,
  showResult: PropTypes.bool.isRequired,
  onSelect: PropTypes.func.isRequired,
  testID: PropTypes.string
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 8,
    padding: 16,
    marginVertical: 8,
    marginHorizontal: 16,
    borderWidth: 2,
  },
  default: {
    backgroundColor: '#FFFFFF',
    borderColor: '#E2E8F0',
  },
  selected: {
    backgroundColor: '#EBF8FF',
    borderColor: '#4299E1',
  },
  correct: {
    backgroundColor: '#C6F6D5',
    borderColor: '#48BB78',
  },
  incorrect: {
    backgroundColor: '#FED7D7',
    borderColor: '#F56565',
  },
  text: {
    fontSize: 16,
    fontWeight: '500',
  },
  defaultText: {
    color: '#2D3748',
  },
  selectedText: {
    color: '#2B6CB0',
  },
  correctText: {
    color: '#2F855A',
  },
  incorrectText: {
    color: '#C53030',
  },
});

export default AnswerOption;
