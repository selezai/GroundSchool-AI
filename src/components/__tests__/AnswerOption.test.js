/* global describe, beforeEach, it, expect, jest */

import React from 'react';
import renderer, { act } from 'react-test-renderer';
import AnswerOption from '../AnswerOption';

describe('AnswerOption Component', () => {
  const defaultProps = {
    option: '1,500 feet',
    isSelected: false,
    isCorrect: false,
    showResult: false,
    onSelect: jest.fn()
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly in default state', () => {
    let tree;
    act(() => {
      tree = renderer.create(<AnswerOption {...defaultProps} />).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when selected', () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <AnswerOption {...defaultProps} isSelected={true} />
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when showing correct answer', () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <AnswerOption 
          {...defaultProps} 
          isCorrect={true} 
          showResult={true} 
        />
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when showing incorrect answer', () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <AnswerOption 
          {...defaultProps} 
          isSelected={true} 
          showResult={true} 
        />
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  it('calls onSelect when pressed', () => {
    let component;
    act(() => {
      component = renderer.create(<AnswerOption {...defaultProps} />);
    });
    
    const button = component.root.findByProps({ testID: 'answer-option' });
    act(() => {
      button.props.onPress();
    });
    
    expect(defaultProps.onSelect).toHaveBeenCalledTimes(1);
  });

  it('disables touch events when showing result', () => {
    let component;
    act(() => {
      component = renderer.create(
        <AnswerOption {...defaultProps} showResult={true} />
      );
    });
    
    const button = component.root.findByProps({ testID: 'answer-option' });
    expect(button.props.disabled).toBe(true);
  });
});
