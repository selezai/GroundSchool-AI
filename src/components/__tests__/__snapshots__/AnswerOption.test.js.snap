// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerOption Component renders correctly in default state 1`] = `null`;

exports[`AnswerOption Component renders correctly when selected 1`] = `null`;

exports[`AnswerOption Component renders correctly when showing correct answer 1`] = `null`;

exports[`AnswerOption Component renders correctly when showing incorrect answer 1`] = `null`;
